tracer/src/Datadog.Trace/Telemetry/TelemetrySettings.cs

```diff
@@ -131,13 +139,37 @@ public static TelemetrySettings FromSource(IConfigurationSource source, IConfigu
                                    .Value;
 
             var dependencyCollectionEnabled = config.WithKeys(ConfigurationKeys.Telemetry.DependencyCollectionEnabled).AsBool(true);
+            // Currently disabled, will be flipped to true in later versions as part of the rollout
+            var v2Enabled = config.WithKeys(ConfigurationKeys.Telemetry.V2Enabled).AsBool(false);
+
+            // Currently disabled, will be flipped to true in later versions as part of the rollout
+            // Also, will require v2 enabled
+            var metricsEnabled = config
+                                .WithKeys(ConfigurationKeys.Telemetry.MetricsEnabled)
+                                .AsBool(
+                                     defaultValue: false,
+                                     validator: enabled =>
+                                     {
+                                         if (v2Enabled || !enabled)
+                                         {
+                                             return true;
+                                         }
+
+                                         configurationError = configurationError is null
```

Comment:
I'm noticing it just now, but it should probably be a helper method (or declaring `configurationErrors` as a list).

Unrelated to this PR though.

Reply:
Yeah, I agree 👍